arch/arm/core/aarch32/cortex_m/relay_vector_table.ld

```diff
@@ -6,5 +6,3 @@
 
 KEEP(*(.vector_relay_table))
 KEEP(*(".vector_relay_table.*"))
-KEEP(*(.vector_relay_handler))
-KEEP(*(".vector_relay_handler.*"))
```

Comment:
I have removed this section and put the `__vector_relay_handler` in `.text`. That way, the two vector tables are one after another and it is more alignment-friendly.

I still have one unresolved problem however, and I'm not sure how to approach this.

The relay vector table gets put at the beginning, so it is always aligned. That is not the case for the real vector table of the application. The alignment is dependent on the number of interrupt handlers.

From Cortex -M4 Devices Generic User Guide

>When setting TBLOFF, you must align the offset to the number of exception entries in the vector
table. The minimum alignment is 32 words, enough for up to 16 interrupts. For more interrupts,
adjust the alignment by rounding up to the next power of two. For example, if you require 21
interrupts, the alignment must be on a 64-word boundary because the required table size is 37
words, and the next power of two is 64.

In `vector_table.ld` how would I have the `exc_vector_table` section be aligned to the nearest power of two depending on its size?

Reply:
@e-rk can you file an issue about the vector table not being aligned?
I think this is a generic issue that we need to address, regardless of whether we do interrupt forwarding or not.

Reply:
Nevermind, I posted a PR that takes care of this (needs review, first but you can try it and see it if solves your problem)
https://github.com/zephyrproject-rtos/zephyr/pull/26800

Reply:
Thank you for doing this. It solves my problem. I'll remember now that Kconfig defines can be also used in ld scripts.